File: src/main/java/com/github/xumeng/java/problems/easy/problem0258/Solution0258.java
package com.github.xumeng.java.problems.easy.problem0258;

public class Solution0258 {

    public int addDigits(int num) {
        if (num < 10) {
            return num;
        }
        int sum = 0;
        for (int i = 9, divisor; i >= 0; i--) {
            divisor = (int) Math.pow(10.0, i);
            sum += (num / divisor);
            num %= divisor;
        }
        while (sum > 9) {
            sum = addDigits(sum);
        }
        return sum;
    }

}

File: src/main/kotlin/com/github/xumeng/kotlin/problems/easy/problem0258/Solution0258.kt
package com.github.xumeng.kotlin.problems.easy.problem0258

import kotlin.math.pow

class Solution0258 {

    fun addDigits(num: Int): Int {
        if (num < 10) {
            return num
        }
        var number = num
        var index = 9
        var divisor: Int
        var quotient: Byte
        var sum = 0
        while (index >= 0) {
            divisor = 10.0.pow(index).toInt()
            quotient = (number / divisor).toByte()
            sum += quotient
            number %= divisor
            index--
        }
        while (sum > 9) {
            sum = addDigits(sum)
        }
        return sum
    }

}